nasa/harmony-gdal-adapter
# HARMONY-1247 - Update to harmony-service-lib==1.0.21.

## Description
This PR ensures the correct error messages are propagated to Harmony from the service container by updating to `harmony-service-lib==1.0.21`.

In v1.0.21, the `harmony-service-lib` ensured that the `error.json` file was [written to S3](https://github.com/nasa/harmony-service-lib-py/blob/main/harmony/cli.py#L150). In v1.0.20 this was not happening. That meant that [the Harmony service-runner](https://github.com/nasa/harmony/blob/main/tasks/service-runner/app/service/service-runner.ts#L71) was looking for `error.json` in S3 and not finding it, thus raising the default `Unknown error` message.

## Jira Issue ID
HARMONY-1247

## Local Test Steps
* Pull this version of HGA.
* Build locally with `./bin/build-image`.
* Restart the services in your local Harmony instance with the `bin/restart-services` script in the Harmony repository.
* Run the [HGA regression test suite](https://github.com/nasa/harmony-regression-tests/blob/main/test/hga/HGA_Regression.ipynb) (or just Sentinel test-case 13, if you want to verify quickly).

## PR Acceptance Checklist
* ~~Jira ticket acceptance criteria met.~~
* ~~Tests added/updated and passing.~~
* ~~Documentation updated (if needed).~~
